packages/shared/src/hooks/streaks/useStreakMilestone.ts

```diff
@@ -36,6 +41,12 @@ export const useStreakMilestone = (): void => {
       return;
     }
 
+    // since the streaks query is cached and has a staleTime set
+    // this is a good time to invalidate that query
+    queryClient.invalidateQueries({
+      queryKey: generateQueryKey(RequestKey.UserStreak, user),
+    });
```

Comment:
I think if we know the action we prefer to mutate rather.
Which I think we do know here, but it's non blocking. (Do note if you not going to change it, maybe we can log it for backlog optimization)

Reply:
As discussed in slack, this is not applicable here unfortunately since we are not relying on a mutation triggered by the client to update streaks, instead these are updated async via the `viewPost` mutation.